server/database/models/article.js

```diff
@@ -82,6 +83,119 @@ export default (sequelize, DataTypes) => {
     },
     {}
   );
+
+  Article.getSearchVector = () => '_search';
+
+  Article.search = async (query, limit, offset) => {
+    query = sequelize.getQueryInterface().escape(query);
+
+    const queryResult = await sequelize.query(
+      `
+      SELECT "Article".*, "comments"."comment" AS "comment",
+      "comments->author"."firstName" AS "firstName", 
+      "comments->author"."lastName" AS "lastName", 
+      "comments->author"."userName" AS "userName", 
+      "comments->author"."avatarUrl" AS "avatarUrl" 
+      FROM (SELECT "Article"."id", "Article"."slug", "Article"."title", 
+      "Article"."description", "Article"."image", "Article"."articleBody", 
+      "Article"."likesCount", "Article"."dislikesCount", 
+      "Article"."publishedAt" FROM "Articles" AS "Article" WHERE 
+      "${Article.getSearchVector()}" @@ plainto_tsquery('english', ${query})
+      AND "Article"."isArchived" = false AND "Article"."publishedAt" IS NOT NULL
+      LIMIT ${limit} OFFSET ${offset}) 
+      AS "Article" LEFT OUTER JOIN "Comments" AS "comments" 
+      ON "Article"."id" = "comments"."articleId" LEFT OUTER JOIN
+      "Users" AS "comments->author" 
+      ON "comments"."userId" = "comments->author"."id"
+      `
+    );
+
+    if (!queryResult[0].length) return { count: 0, rows: [] };
+
+    const formattedResponse = queryResult[0].map((item, index, array) => {
+      const comments = [];
+      if (item.comment) {
+        comments.push({
+          comment: item.comment,
+          author: {
+            firstName: item.firstName,
+            lastName: item.lastName,
+            userName: item.userName,
+            avatarUrl: item.avatarUrl
+          }
+        });
+      }
+      delete item.comment;
```

Comment:
can we make lines 128 - 131 more idiomatic?, something like: `['comment', 'firstname', 'lastname', 'username'].forEach(prop => delete item[prop])`

Reply:
done